src/indexing.jl

```diff
@@ -10,39 +10,53 @@ Base.@propagate_inbounds function getindex(sa::HybridArray{S}, inds::Union{Int,
     _getindex(all_dynamic_fixed_val(S, inds...), sa, inds...)
 end
 
-Base.@propagate_inbounds function _getindex(::Val{:dynamic_fixed_true}, sa::HybridArray, inds::Union{Int, StaticArray{<:Tuple, Int}, Colon}...)
+@inline function Base._getindex(l::IndexLinear, sa::HybridArray{S}, s::Base.Slice) where S
```

Comment:
Do you think it would be good to include some basic information about the indexing process in `Base` here to explain why this is done (and that it relies on undocumented behavior that can change from release to release)?

Reply:
Yes, that's a good idea.